python/lsst/ctrl/oods/msgQueue.py

```diff
@@ -0,0 +1,104 @@
+# This file is part of ctrl_oods
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import asyncio
+import concurrent
+import logging
+
+from confluent_kafka import Consumer
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MsgQueue(object):
+    """Report on new messages
+
+    Parameters
+    ----------
+    config: `dict`
+        configuration dictionary for a consumer
+    topics: `list`
+        The topics to listen on
+    """
+
+    def __init__(self, brokers, group_id, topics, max_messages):
+        self.brokers = brokers
+        self.group_id = group_id
+        self.topics = topics
+        self.max_messages = max_messages
+
+        self.msgList = list()
+        self.condition = asyncio.Condition()
+
+        config = {
+            "bootstrap.servers": ",".join(self.brokers),
+            "group.id": self.group_id,
+            "auto.offset.reset": "earliest",
+        }
+        # note: this is done because mocking a cimpl is...tricky
+        self.createConsumer(config, topics)
+
+    def createConsumer(self, config, topics):
+        """Create a Kafka Consumer"""
+        self.consumer = Consumer(config)
+        self.consumer.subscribe(topics)
+
+    async def queue_files(self):
```

Comment:
I think a lot of this logic could be simplified if you used `aiokafka` instead of `confluent_kafka`, but I understand the latter is more standard.

Reply:
Noted